invision-app/app/components/movieCard/movieCard.js

```diff
@@ -19,39 +21,32 @@ export function MovieCard(props) {
     setHoverState(false);
     setMenuVisibility(false);
   }
+  const closeMenu= useCallback(() => setMenuVisibility(false));
 
-  function openEditWindow() {
-    setAddEditWindowVisibility(true);
-    closeMenu();
-  }
+  const openDeleteWindow = useCallback(() => { setDeleteWindowVisibility(true);setMenuVisibility(false);});
 
-  function closeEditWindow() {
-    setAddEditWindowVisibility(false);
-  }
+  const closeDeleteWindow = useCallback(() => setDeleteWindowVisibility(false));
 
-  function openDeleteWindow() {
-    setDeleteWindowVisibility(true);
-    closeMenu();
-  }
+  const openEditWindow = useCallback(() =>{ setAddEditWindowVisibility(true);setMenuVisibility(false)});
+
+  const closeEditWindow = useCallback(() => setAddEditWindowVisibility(false));
 
-  function closeDeleteWindow() {
-    setDeleteWindowVisibility(false);
-  }
-  function closeMenu(){
-    setMenuVisibility(false);
-  }
   function showDetails(){
-    props.showMovieDetails(props.movie);
+    dispatch(showDescriptionMovie(props.movie));
+   props.showMovieDetails();
+  }
+  function editMovie(movie){
+dispatch(updateMovie(movie));
   }
   const movie = props.movie;
   
   return (
     <article className="movie" onMouseEnter={handleEnter} onMouseLeave={handleLeave}>
       {isOpenedDeleteWindow && (
-        <DeleteMovie movie={props.movieId} close={closeDeleteWindow} deleteMovie={props.deleteMovie} id={props.movie.id}/>
+        <DeleteMovie close={closeDeleteWindow} id={movie.id}/>
```

Comment:
DeleteMovie should have deleteMovie property - error in console: props.deleteMovie is not a function